lib/bloc/marketplace/marketplace_bloc.dart

```diff
@@ -8,13 +8,25 @@ class MarketplaceBloc {
   final _vendorController = BehaviorSubject<List<VendorModel>>();
   Stream<List<VendorModel>> get vendorsStream => _vendorController.stream;
 
+  bool isVendorToggle = true;
```

Comment:
Why are you adding this toggle for snort?

Reply:
Can we move this option to `MarketplaceBloc`? We're using BLoC pattern on breezmobile and this and it's related logic belongs in `MarketplaceBloc` imo.

We could create a `NostrSettings`* class such as:
*: @roeierez or `MarketplaceSettings` ?


```
class NostrSettings {
  final bool showSnort;

  NostrSettings({this.showSnort = true});

  NostrSettings.start() : this(showSnort: true);
  
  NostrSettings copyWith({
    bool showSnort,
  }) {
    return NostrSettings(
      showSnort: showSnort ?? this.showSnort,
    );
  }

  NostrSettings.fromJson(Map<String, dynamic> json)
      : this(showSnort: json["showSnort"] ?? true);

  Map<String, dynamic> toJson() => {
        "showSnort": showSnort,
      };
}
```

_(in case we want to change default behavior or include user preference in backup and one other use case I'll explain on a later step)_
Then later we can expose it from `MarketplaceBloc` via `Stream`/`Sink`:

```
final _nostrSettingsController = BehaviorSubject<NostrSettings>();
Stream<NostrSettings> get nostrSettingsStream =>
    _nostrSettingsController.stream;
Sink<NostrSettings> get nostrSettingsSettingsSink =>
    _nostrSettingsController.sink;
```

to be able to toggle between show/hide.

```
void _toggleSnort(
    MarketplaceBloc bloc,
    NostrSettings nostrSettings,
    ) {
  bloc.nostrSettingsSettingsSink.add(
    nostrSettings.copyWith(
      showSnort: !nostrSettings.showSnort,
    ),
  );
}

```

Each time this setting changes we can update the `NostrSettings` and then reload vendors on the `MarketplaceBloc` by listening to `nostrSettingsStream` such as:

```
  MarketplaceBloc() {
    loadVendors();
    _listenNostrSettings();
  }
  
   _listenNostrSettings() async {
    nostrSettingsStream.listen((settings) async {
      _nostrSettingsController.add(settings);
      await loadVendors();
    });
  }
```